prediction_market_agent/agents/learnable_agent/agents.py

```diff
@@ -0,0 +1,74 @@
+# The code writer agent's system message is to instruct the LLM on how to use
```

Comment:
General Agent is also "learnable agent", probably soemthing like `autogen_agent` is better name?

We could also split it into directory, eg. structure like

```
prediction_market_agent
  agents
    general_agents
      microchain_agent
      autogen_agent
    prediction_market_agents
      arbitrage_agent
      coinflip_agent
      ...
```

Wdyt?

Reply:
Yes, we can do this shuffling on another PR - no strong opinions from my side here
PS - Renamed to `blockchain_coding_agent`